include/eos/core/Landmark.hpp

```diff
@@ -40,12 +41,28 @@ struct Landmark
     LandmarkType coordinates; ///< The position or coordinates of the landmark.
 };
 
+/**
+ * @brief Representation of a landmark with mesh index.
+ */
+template <class LandmarkType>
+struct IndexedLandmark: Landmark<LandmarkType>
```

Comment:
The introduction of a new type `IndexedLandmark` is quite an interesting idea! It would be nice to get a bit of practical experience with it first before merging it into the master branch, and see how useful it would be to the general library, beyond the Ceres fitting (I think it might be quite a nice idea!). It would be nice to put it maybe encapsulated into the Ceres fitting for now?

Reply:
I can just replace it in`ceres_nonlinear.hpp` to not mix this `struct` with non-ceres code. What do you think? Will it be enough?